tests/integration-tests/tests/performance_tests/common.py

```
@@ -0,0 +1,203 @@
+    remote_command_executor,
+    cluster,
+    os,
+    num_compute_nodes,
+    compute_node_instance_type,
+    multithreading_disabled,
+    num_users,
+):
+    test_name = f"{cluster.name}-{num_compute_nodes}nodes"
+    num_jobs_per_iteration = num_compute_nodes
+    num_processes_per_job = fetch_instance_slots(
+        cluster.region, compute_node_instance_type, multithreading_disabled=multithreading_disabled
+    )
+    job_users = [get_username_for_os(os)] + [f"PclusterUser{i}" for i in range(1, num_users)]
+    results_dir = f"/shared/performance-tests/results/{test_name}"
+    log_file = f"{test_name}.log"
+    pid_file = f"{test_name}.pid"
+
+    performance_test_launch_command = f"""
+    '{TEST_RUNNER_SCRIPT}' '{NUM_ITERATIONS}' '{num_jobs_per_iteration}' '{num_processes_per_job}' '{','.join(job_users)}' '{JOB_COMMAND}' '{results_dir}' > '{log_file}' 2>&1 &
+    echo $! > '{pid_file}'
+    cat '{pid_file}'
+    """
+
+    result = remote_command_executor.run_remote_command(performance_test_launch_command, pty=False)
+    assert_that(result.failed).is_false()
+    pid = result.stdout.strip()
+    return pid, results_dir
+
+
+@retry(stop_max_attempt_number=5, wait_fixed=seconds(3))
+def cat_remote_file(remote_command_executor, file_path):
+    logging.info(f"Retrieving remote file {file_path}")
+    result = remote_command_executor.run_remote_command(f"cat {file_path}")
+    assert_that(result.failed).is_false()
+    return result.stdout.strip()
+
+
+@retry(stop_max_attempt_number=60, wait_fixed=seconds(180))
+def wait_for_performance_test_to_complete(remote_command_executor, pid):
+    logging.info("Waiting for performance test to complete")
+    command = f"""
+    ps --pid {pid} > /dev/null
+    if [ "$?" -ne 0 ]; then
+         echo "COMPLETE"
+    else
+        echo "RUNNING"
+    fi
+    """
+    result = remote_command_executor.run_remote_command(command)
+    if result.stdout == "RUNNING":
+        raise "The process is still running"
```

Here you're raising a String, you should raise an exception :D